monopoly/src/main/java/gameset/functionality/Game.java

```diff
@@ -66,21 +66,31 @@ private void listPlayers() {
      * @param player the player object
      * @param property object which the building will be added to
      */
-    private void purchaseBuildings(Property property, Player player) {
-        if (player.canPurchaseBuilding(property.getColor())) {
-            int cost = property.getBuildingPrice();
-            if (player.getMoney() - cost >= 0) {
-                player.setMoney(player.getMoney() - cost);
-                property.buildBuilding();
-                System.out.println("You purchased your " + property.getNumHouses() + " houses of " + property.displayPropertyName());
-            }
-            else {
-                System.out.println("Unable to purchase building due to insufficient funds");
-            }
+    private void purchaseBuildings(Property property, Player player) throws IOException {
+        if (!player.canPurchaseBuilding(property, board)) {
+            System.out.println("Unable to purchase building due to not having all properties of the color set for: " + property.getColor());
+            return;
         }
-        else {
-            System.out.println("Unable to purchase building due to you not having all colorset for: " + property.getColor());
+
+        int cost = property.getBuildingPrice();
+
+        if (property.isMortgaged()) {
+            System.out.println("Unable to purchase buildings on a mortgaged property.");
+            return;
+        }
+
+        if (player.getMoney() >= cost) {
```

Comment:
This looks like it's supposed to be if the cost of building a property is more than player has in balance, then print unable to purchase building. 
```suggestion
        if (cost > player.getMoney()) {
```